src/components/AdvertisingProvider.js

```diff
@@ -3,47 +3,75 @@ import Advertising from '../Advertising';
 import PropTypes from 'prop-types';
 import AdvertisingConfigPropType from './utils/AdvertisingConfigPropType';
 import AdvertisingContext from '../AdvertisingContext';
+import equal from 'fast-deep-equal';
 
 export default class AdvertisingProvider extends Component {
     constructor(props) {
         super(props);
+        this.initialize();
 
-        const { config, plugins } = this.props;
-        this.advertising = new Advertising(config, plugins);
-        this.activate = this.advertising.activate.bind(this.advertising);
-        this.needTearDown = false;
+        const activate = this.advertising.activate.bind(this.advertising);
+        this.state = {
+            activate
+        };
     }
 
-    async setupAdvertising() {
-        await this.advertising.setup();
-        // teardown and setup should run in pair
-        this.needTearDown = true;
```

Comment:
I checked the `Advertising.teardown` function is allow to call without error, even if `Advertising` object is not setup. So I decide to remove this property, to avoid partial setup problem (i.e. Either Prebids or DFP is fail to initialise)